scripts/scil_run_commit.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Estimate the fit between a provided tractogram and DWI. Assign a weight to each
+streamline that represent how well they explain the signal.
+Default is stick-zeppelin-ball for multi-shells data.
+
+The real output from COMMIT is:
+- fit_NRMSE.nii.gz
+    fiting error (Normalized Root Mean Square Error)
+- fit_RMSE.nii.gz
+    fiting error (Root Mean Square Error)
+- results.pickle
+    Dictionary containing the experiment parameters and final weights
+- compartment_EC.nii.gz
+- compartment_IC.nii.gz
+- compartment_ISO.nii.gz
+    Each of COMMIT compartments
+
+This script can divide the input tractogram in two using a threshold to apply
+on the streamlines' weight.
+"""
+
+import argparse
+from contextlib import redirect_stdout
+import io
+import logging
+import os
+import pickle
+import shutil
+import sys
+import tempfile
+
+import commit
+from commit import trk2dictionary
+from dipy.io.stateful_tractogram import (Origin, Space,
+                                         StatefulTractogram)
+from dipy.io.streamline import save_tractogram
+from dipy.io.utils import is_header_compatible
+from dipy.io.gradients import read_bvals_bvecs
+import h5py
+import numpy as np
+import nibabel as nib
+from nibabel.streamlines import Tractogram
+
+from scilpy.io.streamlines import (lazy_streamlines_count,
+                                   reconstruct_streamlines_from_hdf5)
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_processes_arg,
+                             add_verbose_arg,
+                             assert_inputs_exist,
+                             assert_output_dirs_exist_and_empty)
+from scilpy.utils.bvec_bval_tools import fsl2mrtrix, identify_shells
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_tractogram',
+                   help='Input tractogram (.trk or .tck or .h5).')
+    p.add_argument('in_dwi',
+                   help='Diffusion from which the fodf were computed.')
+    p.add_argument('in_bvals',
+                   help='Bvals in the FSL format.')
```

Comment:
(.bval)

Reply:
We also need a --tolerance on bvals as we have in other scripts.
COMMIT needs rounded bvals otherwise, this explodes the number of kernels and atoms in the dictionnary. Maybe default = 20